Language.java

```diff
@@ -0,0 +1,179 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.*;
+
+/**
+ * Class representing an invented language with sorted
+ * list of words
+ * @author Natalija
+ */
+public class Language {
+	List<String> sortedWords;
+
+	public Language(List<String> sortedWords) {
+		if(sortedWords==null || sortedWords.size()==0) {
+			throw new IllegalArgumentException();
+		}
+		this.sortedWords = sortedWords;
+	}
+
+	/**
+	 * Creates an alphabet for the invented language
+	 * from an ordered list of words. 
+	 * return - Sorted list representing the alphabet 
+	 */
+	public List<Character> getSortedAlphabet() {
+		
+		//In order to perform topological sort we need to build a graph from the
+		// sorted words of the unknown language
+		Pair<Map<Character, Set<Character>>, Map<Character, Integer>> graph_degree = buildGraph();
+		Map<Character, Set<Character>> graph = graph_degree.getFirst();
+		Map<Character, Integer> degree = graph_degree.getSecond();
+		
+		return topologicalSort(graph, degree);
+	}
+
+
+	/**
+	 * Calculates the graph and the indegrees of it's vertices. 
+	 * 
+	 * For the sorted words [ART, RAT, CAT, CAR] the graph would be: 
+	 * 		A->R 
+	 * 		R->C
+	 * 		T->R
+	 * The indegrees of the vertices would be: 
+	 * 		A: 0
+	 * 		R: 2
+	 * 		C: 1
+	 * 		T: 0
+	 * 
+	 * @return - A pair of two Maps: The first one representing the graph created 
+	 * from the sorted words and the second one representing the indegrees of 
+	 * each vertex of the created graph. 
+	 */
+	private Pair<Map<Character, Set<Character>>, Map<Character, Integer>> buildGraph() {
+		Map<Character, Set<Character>> graph = new HashMap<Character, Set<Character>>();
+		Map<Character, Integer> degree = new HashMap<Character, Integer>();
+		Set<Character> uniqueChars = uniqueCharacters();
+		
+		if(sortedWords.size()==1){
+			String currentWord = sortedWords.get(0);
+			for(char c: currentWord.toCharArray()) {
+				graph.putIfAbsent(c, new HashSet<>());
+				degree.putIfAbsent(c, 0);
+			}
+		} else {
+			for (int i = 0; i < sortedWords.size() - 1; i++) {
+				String currenttWord = sortedWords.get(i);
+				String adjacentWord = sortedWords.get(i + 1);
+	
+				boolean wait = true;
+				int count = 0;
+				int limit = Math.min(currenttWord.length(), adjacentWord.length());
+				//Don't enter if word is empty
+				wait = wait && count < limit;
+				
+				while (wait) {
+					if (currenttWord.charAt(count) != adjacentWord.charAt((count))) {
```

Comment:
pulling currenttWord.charAt(count) out into a local variable would clean this up a lot.